pkg/crypto/certloader/client_test.go

```diff
@@ -124,8 +124,19 @@ func TestWatchedClientConfigRotation(t *testing.T) {
 	assert.NotNil(t, c)
 	defer c.Stop()
 
+	keypairGeneration, caCertPoolGeneration := c.keypairGeneration, c.caCertPoolGeneration
 	rotate(t, hubble, relay)
-	<-time.After(testReloadDelay)
+
+	// wait until both keypair and caCertPool have been reloaded
+	ticker := time.NewTicker(testReloadDelay)
+	defer ticker.Stop()
+	for range ticker.C {
+		keypairUpdated := c.keypairGeneration > keypairGeneration
```

Comment:
Is this thread-safe, i.e. passes `go test -race`? If the generation is updated by a different go routine in the background, I think we need to synchronize the access somehow.

Reply:
ah good catch I'll add getters for them